tests/dependencies/is_into_series_test.py

```diff
@@ -0,0 +1,18 @@
+from __future__ import annotations
```

Comment:
Commenting here to address entire file. On the flow of #1199 I would suggest to use `constructor_eager` for anything returning True, and manually constructor only numpy array or list that return False. Probably the easiest way of achieving that is by splitting into two different tests

Reply:
Makes sense to me. Thanks for the feedback @FBruzzesi! Taking a look

Reply:
I'm having some trouble committing my updates to the tests. Here's what I have, but I'm getting two errors.

```
from __future__ import annotations

import numpy as np

import narwhals as nw
from narwhals.dependencies import is_into_series
from tests.utils import ConstructorEager

def test_is_into_series_true(constructor_eager: ConstructorEager) -> None:
    raw_series = constructor_eager({"a": [1, 2, 3]})["a"]
    assert is_into_series(raw_series)
    nw_series = nw.from_native(raw_series, allow_series=True)
    assert is_into_series(nw_series)
```

One error I'm getting with committing is:

```
tests/dependencies/is_into_series_test.py:7:25: TCH001 Move application import `tests.utils.ConstructorEager` into a type-checking block
```
I can't figure this one out. It looks like I'm importing it in the same way as other tests do. 

The other error I get is:

```
tests/dependencies/is_into_series_test.py:11: error: Value of type "NativeFrame | DataFrame[Any] | DataFrameLike" is not indexable  [index]
Found 1 error in 1 file (checked 1 source file)
```
So I guess this is not always expected to work for `ConstructorEager` type?

```
constructor_eager({"a": [1, 2, 3]})["a"]
```
Or there's something I need to add to the types "NativeFrame", "DataFrame[Any]", "DataFrameLike"?

Reply:
Hey @LiamConnors 

> tests/dependencies/is_into_series_test.py:7:25: TCH001 Move application import `tests.utils.ConstructorEager` into a type-checking block

will require to do:

```py
from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from tests.utils import ConstructorEager
```

This is because of `from __future__ import annotations`


> tests/dependencies/is_into_series_test.py:11: error: Value of type "NativeFrame | DataFrame[Any] | DataFrameLike" is not indexable  [index]
> Found 1 error in 1 file (checked 1 source file)

is because lazy frame does not support indexing. It should not complain if you specify `nw.from_native(..., eager_only=True)`